hal/src/stm32f2xx/deviceid_hal.cpp

```diff
@@ -126,6 +127,20 @@ int hal_get_device_secret(char* data, size_t size, void* reserved)
 }
 #endif /* HAL_DEVICE_ID_NO_DCT */
 
+#ifndef HAL_DEVICE_ID_NO_DCT
+int hal_get_device_ncp_id(uint8_t * ncpid)
+{
+    uint8_t otp_ncp_id;
+    int ret = FLASH_ReadOTP(DEVICE_NCP_ID_OTP_ADDRESS, (uint8_t*)&otp_ncp_id, sizeof(otp_ncp_id));
```

Comment:
`FLASH_ReadOTP` may very well return garbage for devices with unpopulated OTP area, the output should be sanitized, hence the above proposal to instead implement `platform_ncp_primary_identifier()` which provides a concrete NCP identifier or `PLATFORM_NCP_UNKNOWN`.

Reply:
moved this logic to `platform_ncp_electron.cpp`